python/wasmedge_sock/wasi_socket_ext.c

```diff
@@ -700,71 +701,56 @@ int getpeername(int sockfd, struct sockaddr* restrict addr,
   return 0;
 }
 
-int getsockname(int fd, struct sockaddr* restrict addr,
+int getsockname(int sockfd, struct sockaddr* restrict addr,
                 socklen_t* restrict addrlen) {
-  // addrlen contains the size of the struct pointed at by addr
-  // it needs to be returned with the actual size of the address
-
-  if (addr == NULL || addrlen == NULL) {
```

Comment:
I think we still need a null check to match the behavior of linux, and also because both of them get dereferenced.